### PR TITLE
goaway: log server addr in client logs

### PR DESCRIPTION
**Pull request by purnesh42H**
Fixes: #7265

RELEASE NOTES: None

### CODE REVIEW

File: internal/transport/keepalive_test.go
Comment (arjan-bal) [inline internal/transport/keepalive_test.go:450]: nit: %s instead of %v.

**Review by arjan-bal — APPROVED** [review]

**Comment by aranjans** [comment]
LGTM!

**Review by aranjans — APPROVED** [review]

**Comment by arvindbr8** [comment]
Thanks for taking a stab at this @purnesh42H. 

FYI: We had a user send a similar themed PR https://github.com/grpc/grpc-go/pull/7150.  After discussing this with team  - I have a [branch](https://github.com/grpc/grpc-go/compare/master...arvindbr8:grpc-go:improve_transport_logs) which aims to get the logging better with remote addr for the whole of the transport package.  Along with adding more info to the log, I'm also attempted at unexporting  transport.ConnectionError. This opened up a can of worms and still fixing some of the tests. 

How do you feel about waiting on me to get that fix completed? I think that patchset should also tackle what's done in this PR.

**Comment by purnesh42H** [comment]
> branch

Sure, yeah that sounds good to me. I will inform the reporter that they can unblock themselves by making similar change locally or wait for your branch to release

**Pull request closed without merging**
